contracts/interchain-token-service/InterchainTokenService.sol

```diff
@@ -398,42 +400,83 @@ contract InterchainTokenService is
         );
     }
 
+    // Returns the amount of token that this call is worth. If `tokenAddress` is `0`, then value is in terms of the native token, otherwise it's in terms of the token address.
+    function contractCallValue(
+        string calldata sourceChain,
+        string calldata sourceAddress,
+        bytes calldata payload
+    ) public view virtual onlyRemoteService(sourceChain, sourceAddress) notPaused returns (address, uint256) {
+        (, bytes32 tokenId, , uint256 amount) = abi.decode(payload, (uint256, bytes32, bytes, uint256));
```

Comment:
this should have the same checks apply, such as revert on invalid selector since that is not express callable. The relayer will query this method to determine if it can be expressed

Reply:
Specifically deploying tokens should NOT be expressable. Only receive token and receive token with data should.

Reply:
It was actually implemented in the `expressExecute` method

Reply:
yeah, but we want to fail here as well so the relayer knows before they even try doing expressExecute

Reply:
Added